### PR TITLE
Emit BidCancelled/BidPlaced Events, Azure Event Hub Integration

### PR DESCRIPTION
**Pull request by RyanCross**
Merging @JamesEarle's Eventhub integration work, fixed some minor linter warnings and added message version to message schemas

### CODE REVIEW

**Comment by RyanCross** [comment]
@JamesEarle if you update circle-ci with the new event hub params, the tests should pass. (Or i can do it if you send me the login)

File: src/api.ts [hunk 1]
```diff
@@ -188,6 +204,18 @@ router.post(
       // Add new bid document to database
       await database.insertBid(newBid);
 
+      const message: BidPlacedMessage = {
+        event: "BidPlaced",
+        version: "1.0",
+        timestamp: new Date().getTime(),
+        logIndex: null,
+        blockNumber: null,
```
Comment (JamesEarle) [inline src/api.ts:212]: We need to have these values for tracking historically. Block number tells us which block an event happened on, but there is no guarantee for order of events being read from our queue so we use `logIndex` to track it simply as `0, 1, 2...`

You can get block number using an existing utility in `src > utils > contracts`
```
  const ethersProvider = getEthersProvider();
  const blockNum = ethers.BigNumber.from(await ethersProvider.getBlockNumber());
```
Don't exactly remember how we created logIndex in the past, let me see if I can find reference to this

File: src/messagequeue/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { BidCancelledMessage, BidPlacedMessage } from "../types";
+
+export * from "./adapters";
+
+export interface MessageQueueService {
+  sendMessage: (message: BidPlacedMessage | BidCancelledMessage) => Promise<void>;
```
Comment (JamesEarle) [inline src/messagequeue/index.ts:6]: no `BidAccepted` ?

File: src/types.ts [hunk 3]
```diff
@@ -73,6 +73,26 @@ export interface VerifyBidResponse {
...
+
+export interface BidCancellation {
+  account: string;
+  auctionId: string;
+}
+
+export interface BaseMessage {
+  event: string;
+  timestamp: number;
+  logIndex: null;
```
Comment (JamesEarle) [inline src/types.ts:88]: again we need these numbers for disambiguation, they shouldn't be null

**Review by JamesEarle — COMMENTED** [review]
A few comments

File: src/types.ts (hunk 3 above)
Comment (JamesEarle) [reply src/types.ts:88]: nvm

**Review by Remscar — APPROVED** [review]

**Pull request merged**